### PR TITLE
Swagger

### PR DESCRIPTION
**Pull request by jcgrethe**
## Summary
- Added swagger doc for Book model and controller
## Screenshots
Screenshot from 2019-07-30 15-51-53

### CODE REVIEW

**Review by Gianni-Nicolas — APPROVED** [review]

File: src/main/java/wolox/training/config/swaggerConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    public Docket productApi() {
+        return new Docket(DocumentationType.SWAGGER_2).select()
+            .apis(RequestHandlerSelectors.any())
+            .paths(PathSelectors.any())
+            .build().apiInfo(apiEndPointInfo());
+    }
+
+    private ApiInfo apiEndPointInfo(){
+        return new ApiInfoBuilder().title("Wolox Training")
+            .description("REST API for libraries")
```
Comment (GNico95) [inline src/main/java/wolox/training/config/swaggerConfiguration.java]: I recommend changing the default swagger message in .description (). You could put a description of what your application does.

**Review by Gianni-Nicolas — DISMISSED** [review]

File: src/main/java/wolox/training/controllers/BookController.java [hunk 2]
```diff
@@ -24,25 +30,49 @@
...
         bookRepository.deleteById(id);
     }
 
     @PutMapping("/{id}")
-    public Book update(@RequestBody Book book, @PathVariable Long id)
+    @ApiOperation(value = "update a book", response = Book.class)
+    @ApiResponses(
+        value = {@ApiResponse(code = 200, message = "Book updated"),
+            @ApiResponse(code = 404, message = "Book not found"),
+            @ApiResponse(code = 404, message = "Bad id")
```
Comment (Jimenarosello) [inline src/main/java/wolox/training/controllers/BookController.java]: I think the code here should be 400 (BookMismatchException?)

**Review by Jimenarosello — CHANGES_REQUESTED** [review]
Looks good, just one small change :)

File: src/main/java/wolox/training/controllers/BookController.java (hunk 2 above)
Comment (jcgrethe) [reply src/main/java/wolox/training/controllers/BookController.java]: yes, my mistake

**Review by Jimenarosello — APPROVED** [review]

**Review by Gianni-Nicolas — APPROVED** [review]

**Pull request merged**